BUILDING.md

```diff
@@ -0,0 +1,39 @@
+# Building Protocol Buffers
+
+These are instructions for building the open source release of protobuf from source.
+
+Given the right tools, this project can be built on Linux, Mac OS X, or Windows though Linux is by far the most commonly used and best tested platform.
+
+## Prerequisites
+
+Likely incomplete
+
+* git
+* JDK 8 or JDK 11.
+* bazel
+* clang
+
+At the top level, this project is built with bazel. Java parts 
+are built with Maven. All languages assume that protoc is built and
+installed first. 
+
+
+## Detailed instructions
+
+1. `$ git clone git@github.com:protocolbuffers/protobuf.git`. You can also clone the repo over https.
+1. `$ cd protobuf`
+1. `$ git clean -xfd .` (Important for cleaning up after previous builds.)
+1. `$ bazel build //...`
```

Comment:
I think if we want these explicit instructions we should provide targets for all the languages to build

Reply:
Ideally, there's a single command that encompasses all subcommands. If that proves not to be true, then yes, we need to add more commands here or even better improve the top-level build to run all the necessary subcommands.

Reply:
It is very rare that a person would want to build the whole protobuf library. More likely, they want to pick a single language and either full or lite and then maybe protoc. So this should have specific build targets for those common workflows, instead of a generic all-encompassing build.